tests/loggers/test_all.py

```diff
@@ -365,6 +365,16 @@ def _test_logger_created_on_rank_zero_only(tmpdir, logger_class):
     assert trainer.state.finished, f"Training failed with {trainer.state}"
 
 
+def test_none_logger(tmpdir):
+    """Tests the case where logger=None is passed to Trainer."""
+    model = BoringModel()
+    trainer = Trainer(
+        logger=None,
```

Comment:
Technically this should be caught by a type checker since the trainer does explicitly allow for None to be passed in. Logger currently isn't an optional input

Reply:
Is it worth changing the type to `Optional` now? I think `logger=None` is something that people can mistakenly assume support exists for. The type checking warnings must not be obvious enough if this issue is coming up, so it could be simplest to just expand the types.

Reply:
Let's file an issue to discuss this. I think there's precedent for it given profiler accepts an optional input. But let's decouple that from this PR

Reply:
https://github.com/PyTorchLightning/pytorch-lightning/issues/12259